plugins/woo-ai/api/attribute-suggestion/class-attribute-suggestion-api.php

```diff
@@ -0,0 +1,218 @@
+							'validate_callback' => 'rest_validate_request_arg',
+							'sanitize_callback' => 'sanitize_text_field',
+							'default'           => '',
+						),
+						'description'         => array(
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+							'sanitize_callback' => 'sanitize_text_field',
+							'default'           => '',
+						),
+						'categories'          => array(
+							'type'              => 'array',
+							'validate_callback' => 'rest_validate_request_arg',
+							'items'             => array(
+								'type'              => 'string',
+								'sanitize_callback' => 'sanitize_text_field',
+							),
+							'default'           => array(),
+						),
+						'tags'                => array(
+							'type'              => 'array',
+							'validate_callback' => 'rest_validate_request_arg',
+							'items'             => array(
+								'type'              => 'string',
+								'sanitize_callback' => 'sanitize_text_field',
+							),
+							'default'           => array(),
+						),
+						'other_attributes'    => array(
+							'type'              => 'array',
+							'validate_callback' => 'rest_validate_request_arg',
+							'default'           => array(),
+							'items'             => array(
+								'type'       => 'object',
+								'properties' => array(
+									'key'   => array(
+										'type' => 'string',
+										'sanitize_callback' => 'sanitize_text_field',
+									),
+									'value' => array(
+										'type' => 'string',
+										'sanitize_callback' => 'sanitize_text_field',
+									),
+								),
+							),
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Check if a given request has access to create a product.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|boolean
+	 */
+	public function get_response_permission_check( WP_REST_Request $request ) {
+		if ( ! wc_rest_check_post_permissions( 'product', 'create' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot view this resource.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the attribute suggestions.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_HTTP_Response|WP_REST_Response
+	 */
+	public function get_response( WP_REST_Request $request ) {
+		$requested_attribute = $request->get_param( 'requested_attribute' );
+		$name                = $request->get_param( 'name' );
+		$description         = $request->get_param( 'description' );
+		$categories          = $request->get_param( 'categories' );
+		$tags                = $request->get_param( 'tags' );
+		$attributes          = $request->get_param( 'attributes' );
+
+		// Strip HTML tags from the description.
+		if ( ! empty( $description ) ) {
+			$description = wp_strip_all_tags( $request->get_param( 'description' ) );
```

Comment:
It's great to see that we're stripping out tags to make sure there's less risk of prompt injection/weirdness being sent to the text completion service. Should we be stripping tags from all of the params, not just the description?

Reply:
Hmm. We can certainly do that, but then if the title has a decorative string that looks like a tag like "A shiny shoe < Now in Black >" we might lose that part. I'm not sure if removing the tags will help with prompt injection, though since it [seems possible plain text too](https://simonwillison.net/2023/Apr/14/worst-that-can-happen/).